### PR TITLE
Update to Ember 3.16

### PR DESCRIPTION
**Pull request by BarryThePenguin**
Was looking at rollbar today, and noticed a message at the top of our dashboard

rollbar

So I thought I'd put together another upgrade PR 😅 

I ran `ember-cli-update --to 3.16.2`

And fixed the lint issues, most of which were released in [`eslint-plugin-ember@8.0.0`](https://github.com/ember-cli/eslint-plugin-ember/releases/tag/v8.0.0)

### CODE REVIEW

File: addon/services/rollbar.js [hunk 1]
```diff
@@ -5,56 +5,63 @@ import deepMerge from 'lodash.merge';
 import Rollbar from 'rollbar';
 
 export default Service.extend({
-  enabled: computed({
+  _rollbar: null,
+
+  enabled: computed('config.enabled', {
```
Comment (BarryThePenguin) [inline addon/services/rollbar.js:10]: See [require-computed-property-dependencies](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/require-computed-property-dependencies.md)

File: addon/services/rollbar.js [hunk 2]
```diff
@@ -5,56 +5,63 @@ import deepMerge from 'lodash.merge';
...
     get() {
-      return this.get('config.enabled');
+      return this.config.enabled;
```
Comment (BarryThePenguin) [inline addon/services/rollbar.js:12]: See [no-get](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md)

File: addon/services/rollbar.js [hunk 3]
```diff
@@ -5,20 +5,20 @@ import deepMerge from 'lodash.merge';
 import Rollbar from 'rollbar';
 
 export default Service.extend({
-  enabled: computed({
+  enabled: computed('config.enabled', {
     get() {
-      return this.get('config.enabled');
+      return this.config.enabled
```
Comment (BarryThePenguin) [inline addon/services/rollbar.js:10]: See [no-get](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md)

File: tests/dummy/app/components/scroll-to/component.js [hunk 4]
```diff
@@ -7,8 +7,8 @@ export default Component.extend({
 
   click() {
     next(() => {
-      let element = document.querySelector(this.get('href'));
-      let position = element.offsetTop + this.get('offset');
+      let element = document.querySelector(this.element.hash);
```
Comment (BarryThePenguin) [inline tests/dummy/app/components/scroll-to/component.js:10]: I wasn't able to find out why.. but the `this.href` property isn't available on the component anymore. I was able to use `this.element.hash`

Maybe it's glimmer related? Something about the difference between html attributes, and component properties

**Comment by camerondubas** [comment]
Was also just looking at rollbar and notices that message. Would be great to get this and #51 Merged 👍

File: addon/services/rollbar.js (hunk 1 above)
Comment (Exelord) [reply addon/services/rollbar.js:8]: That was intentional :) We should revert.

File: addon/services/rollbar.js (hunk 3 above)
Comment (Exelord) [reply addon/services/rollbar.js:10]: That was for compatibility issues :) Can we revert all changes to files not related to update?

File: config/ember-try.js [hunk 5]
```diff
@@ -6,18 +6,18 @@ module.exports = async function() {
   return {
     scenarios: [
       {
-        name: 'ember-lts-3.8',
+        name: 'ember-lts-3.12',
```
Comment (Exelord) [inline config/ember-try.js:9]: I still want to support Ember 3.8. That's a huge breaking change :/

File: package.json [hunk 6]
```diff
@@ -16,60 +16,61 @@
...
   "devDependencies": {
-    "@ember/optional-features": "^1.1.0",
+    "@ember/optional-features": "^1.3.0",
     "@fortawesome/ember-fontawesome": "^0.2.1",
-    "@fortawesome/free-brands-svg-icons": "^5.12.0",
-    "@glimmer/component": "^1.0.0",
-    "babel-eslint": "^10.0.3",
+    "@fortawesome/free-brands-svg-icons": "^5.14.0",
+    "@glimmer/component": "^1.0.1",
+    "@glimmer/tracking": "^1.0.0",
```
Comment (Exelord) [inline package.json:39]: Why tracking landed here? :)

File: tests/dummy/app/components/scroll-to/component.js (hunk 4 above)
Comment (Exelord) [reply tests/dummy/app/components/scroll-to/component.js:10]: Its dummy app :) FIne for me

File: package.json (hunk 6 above)
Comment (BarryThePenguin) [reply package.json:39]: It's added automatically when running `ember-cli-update --to 3.16.2`

I can remove it?

File: config/ember-try.js (hunk 5 above)
Comment (BarryThePenguin) [reply config/ember-try.js:9]: I can remove `@glimmer/tracking` and revert the ember-try config

This was all automatically done when running `ember-cli-update --to 3.16.2`

There's no need to drop support if there are no breaking changes 👀

**Comment by BarryThePenguin** [comment]
I've removed the code changes that were in this PR

**Pull request closed without merging**